rotkehlchen/chain/ethereum/modules/makerdao/decoder.py

```diff
@@ -557,31 +571,31 @@ def decode_saidai_migration(self, context: DecoderContext) -> DecodingOutput:
-            string_to_evm_address('0x7f62f9592b823331E012D3c5DdF2A7714CfB9de2'): (self.decode_makerdao_vault_action, A_WBTC.resolve_to_crypto_asset(), 'WBTC-C'),  # noqa: E501
-            string_to_evm_address('0xc7e8Cd72BDEe38865b4F5615956eF47ce1a7e5D0'): (self.decode_makerdao_vault_action, A_ZRX.resolve_to_crypto_asset(), 'ZRX-A'),  # noqa: E501
-            string_to_evm_address('0xA6EA3b9C04b8a38Ff5e224E7c3D6937ca44C0ef9'): (self.decode_makerdao_vault_action, A_MANA.resolve_to_crypto_asset(), 'MANA-A'),  # noqa: E501
-            string_to_evm_address('0x7e62B7E279DFC78DEB656E34D6a435cC08a44666'): (self.decode_makerdao_vault_action, A_PAX.resolve_to_crypto_asset(), 'PAXUSD-A'),  # noqa: E501
-            string_to_evm_address('0xBEa7cDfB4b49EC154Ae1c0D731E4DC773A3265aA'): (self.decode_makerdao_vault_action, A_COMP.resolve_to_crypto_asset(), 'COMP-A'),  # noqa: E501
-            string_to_evm_address('0x6C186404A7A238D3d6027C0299D1822c1cf5d8f1'): (self.decode_makerdao_vault_action, A_LRC.resolve_to_crypto_asset(), 'LRC-A'),  # noqa: E501
-            string_to_evm_address('0xdFccAf8fDbD2F4805C174f856a317765B49E4a50'): (self.decode_makerdao_vault_action, A_LINK.resolve_to_crypto_asset(), 'LINK-A'),  # noqa: E501
-            string_to_evm_address('0x4a03Aa7fb3973d8f0221B466EefB53D0aC195f55'): (self.decode_makerdao_vault_action, A_BAL.resolve_to_crypto_asset(), 'BAL-A'),  # noqa: E501
-            string_to_evm_address('0x3ff33d9162aD47660083D7DC4bC02Fb231c81677'): (self.decode_makerdao_vault_action, A_YFI.resolve_to_crypto_asset(), 'YFI-A'),  # noqa: E501
-            string_to_evm_address('0xe29A14bcDeA40d83675aa43B72dF07f649738C8b'): (self.decode_makerdao_vault_action, A_GUSD.resolve_to_crypto_asset(), 'GUSD-A'),  # noqa: E501
-            string_to_evm_address('0x3BC3A58b4FC1CbE7e98bB4aB7c99535e8bA9b8F1'): (self.decode_makerdao_vault_action, A_UNI.resolve_to_crypto_asset(), 'UNI-A'),  # noqa: E501
-            string_to_evm_address('0xFD5608515A47C37afbA68960c1916b79af9491D0'): (self.decode_makerdao_vault_action, A_RENBTC.resolve_to_crypto_asset(), 'RENBTC-A'),  # noqa: E501
-            string_to_evm_address('0x24e459F61cEAa7b1cE70Dbaea938940A7c5aD46e'): (self.decode_makerdao_vault_action, A_AAVE.resolve_to_crypto_asset(), 'AAVE-A'),  # noqa: E501
-            string_to_evm_address('0x885f16e177d45fC9e7C87e1DA9fd47A9cfcE8E13'): (self.decode_makerdao_vault_action, A_ETH_MATIC.resolve_to_crypto_asset(), 'MATIC-A'),  # noqa: E501
+            string_to_evm_address('0x3D0B1912B66114d4096F48A8CEe3A56C231772cA'): (self.decode_makerdao_vault_action, string_to_evm_address('0x3D0B1912B66114d4096F48A8CEe3A56C231772cA'), A_BAT.resolve_to_crypto_asset(), 'BAT-A'),  # noqa: E501
+            string_to_evm_address(MAKERDAO_GEM_JOIN_ETHA_ADDRESS): (self.decode_makerdao_vault_action, string_to_evm_address(MAKERDAO_GEM_JOIN_ETHA_ADDRESS), A_WETH.resolve_to_crypto_asset(), 'ETH-A'),  # noqa: E501
```

Comment:
I am not sure if simply replacing ETH with WETH works here. It may work for some cases and fail for others.

I think the solution is to work on the `decode_makerdao_vault_action` to treat ETH and WETH the same

Reply:
Without it, events are not correctly decoded when transactions are complex. In this way instead the events are "deterministic" and they can be compared with the logs on blockchain for reviewing. I will push another PR on top of this branch for a fix on the WETH module, just for discussing. Because I prefer to show what I mean using code.